packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -124,22 +153,175 @@ const AccountRecoveryResetPassword = ({
+
+      forceAuth = !!verificationInfo.forceAuth;
+
+      if (!verificationInfo.isValid()) {
+        supportsRecovery = false;
+        linkStatus = 'damaged';
+      } else if (!accountRecoveryKeyInfo.isValid()) {
+        supportsRecovery = false;
+      } else if (accountRecoveryKeyInfo.lostRecoveryKey === true) {
+        supportsRecovery = false;
+      }
+    } catch (err) {
+      if (err instanceof ContextValidationErrors) {
+        contextError = err;
+        linkStatus = 'damaged';
+      }
+    }
+
+    return {
+      email,
+      linkStatus,
+      forceAuth,
+      supportsRecovery,
+      contextError,
+    };
+  }
+
+  async function onSubmit(data: FormData) {
+    const password = data.newPassword;
+
+    try {
+      await account.resetPasswordWithRecoveryKey(
+        {
+          password,
+          ...verificationInfo,
+          ...accountRecoveryKeyInfo
+        }
+      );
+
+      // FOLLOW-UP: I don't see functionality in settings
+      await account.setLastLogin(Date.now());
+
+      // FOLLOW-UP: It seems like the account class will now refresh itself, so I don't think
+      //            these are necessary anymore?
+      // storageContext.set('currentAccountUid', account.uid);
+      // await account.refresh('account');
+
+      // FOLLOW-UP: No equivalent yet in settings
+      notifier.onAccountSignIn(account);
+
+      relier.resetPasswordConfirm = true;
+      logViewEvent(viewName, 'verification.success');
+
+      // FOLLOW-UP: Broker not yet implemented
+      await broker.invokeBrokerMethod('afterCompleteResetPassword', account);
+      alertSuccess();
+      navigateAway();
+    } catch (err) {
+      if (AuthUiErrors['INVALID_TOKEN'].errno === err.errno) {
+        // TODO: Is this needed? Had to add the a method to support this.
+        // BEFORE: this.logError(err);
+        logErrorEvent({ viewName, ...err });
+        setLinkStatus('expired');
+      } else {
+        // Context validation errors indicate a bad state in either the url query or
+        // maybe storage. In these cases show an alert bar and let the error keep bubbling
+        // up.
+        if (err instanceof ContextValidationErrors) {
+          alertInvalidContext(err);
+        } else {
+          logErrorEvent(err);
+          const msg = ftlMsgResolver.getMsg(
+            'unexpected-error-encountered',
+            'Unexpected Error Encountered'
+          );
+          alertBar.error(msg);
+        }
+        throw err;
+      }
+    }
+  }
+
+  async function resendLinkHandler() {
+    logViewEvent(viewName, 'account-recovery-reset-password.resend');
+
+    // FOLLOW-UP: The previous code would retry the resend several times. Is that really necessary...
+    try {
+      await account.resetPassword(state.email);
+      const msg = ftlMsgResolver.getMsg(
+        `email-resent`,
+        'Email resent. Add accounts@firefox.com to your contacts to ensure a smooth delivery.'
+      );
+      alertBar.success(msg);
```

Comment:
I think these success/error messages for resending an email should be displayed in the Banner component? We typically use the alertBar when the message needs to persist to the next page on navigate.

Reply:
Okay, I don't think banner was in place at time of initial PR. I'll check this out though. :thumbsup:

Reply:
That's probably true! FWIW, I also filed a follow-up ticket today to adjust the banner on some reset-password pages, so changing the alert bar to a banner here could be rolled into that ticket as a follow up since this PR is already pretty big. 
https://mozilla-hub.atlassian.net/browse/FXA-7019

Reply:
Oh that would be nice. Just want to get this landed. I'll add some follow up comments, and update that ticket ^.

Reply:
I looked into the Banner. It seems straight forward enough. So I'll be pushing an update shortly.